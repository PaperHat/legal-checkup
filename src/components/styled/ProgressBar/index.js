import styled from 'styled-components'

const ProgressBar = styled.nav`
  display: flex;
`

ProgressBar.displayName = 'StyledProgressBar'

export default ProgressBar
