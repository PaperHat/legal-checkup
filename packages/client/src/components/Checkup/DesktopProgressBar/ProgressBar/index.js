import styled from 'styled-components'

const ProgressBar = styled.nav`
  background-color: #fff;
  display: flex;
  justify-content: center;
  margin: 20px 0;
`

ProgressBar.displayName = 'StyledDesktopProgressBar'

export default ProgressBar
