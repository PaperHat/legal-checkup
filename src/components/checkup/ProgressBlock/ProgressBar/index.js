import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const ProgressBar = styled.nav`
  background-color: #eeedea;
  display: flex;
  width: 100%;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  letter-spacing: normal;
  text-align: center;
  margin: 20px 0;

  ${styledMap('format', {
    [DESKTOP]: ``,
    [MOBILE]: ``,
    [TABLET]: ``
  })}
`

ProgressBar.displayName = 'ProgressBar'

export default ProgressBar
