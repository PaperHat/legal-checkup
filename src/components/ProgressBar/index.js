import React from 'react'

import { StyledProgressBar } from '../styled'
import { CurrentTopic, NavButtonBack, NavButtonForward } from '../../containers'

export default function ProgressBar () {
  return (
    <StyledProgressBar>
      <NavButtonBack>&#x25C0;</NavButtonBack>
      <CurrentTopic />
      <NavButtonForward>&#x25B6;</NavButtonForward>
    </StyledProgressBar>
  )
}
