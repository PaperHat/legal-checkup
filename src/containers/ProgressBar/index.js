import { connect } from 'react-redux'

import { ProgressBar } from '../../components'
import {
  getActiveQuestion,
  getQuestions,
  getResponses,
  questionSelected,
  previousQuestion,
  nextQuestiion
} from '../../state'

function mapStateToProps (state) {
  return {
    activeQuestion: getActiveQuestion(state),
    questions: getQuestions(state),
    responses: getResponses(state)
  }
}

function mapDispatchToProps (dispatch) {
  return {
    handleClick: questionNumber => dispatch(questionSelected(questionNumber)),
    handlePrevious: () => dispatch(previousQuestion()),
    handleNext: () => dispatch(nextQuestiion())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProgressBar)
