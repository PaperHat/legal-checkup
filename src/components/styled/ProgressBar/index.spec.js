import React from 'react'
import { shallow } from 'enzyme'

import ProgressBar from '.'

describe('components:styled:Progressbar', () => {
  it('renders the ProgressBar and CSS properties properly', () => {
    expect(
      toJson(shallow(<ProgressBar>ProgressBar</ProgressBar>))
    ).toMatchSnapshot()
  })
})
