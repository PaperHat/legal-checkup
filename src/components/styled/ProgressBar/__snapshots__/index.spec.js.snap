// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:Progressbar renders the ProgressBar and CSS properties properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<nav
  className="c0"
>
  ProgressBar
</nav>
`;
